### PR TITLE
Optimize the initialization of the edgecore modules global variable

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

 /kind cleanup
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Optimize the initialization of the edgecore modules global variable
```

### CODE REVIEW

File: edge/pkg/common/config/config.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			if len(errs) != 0 {
+				for _, e := range errs {
+					klog.Errorf("%v", e)
+				}
+				klog.Error("init common config error")
+				os.Exit(1)
+			} else {
+				klog.Infof("init common config successfully，config info %++v", c)
+			}
+		}()
```
Comment (fisherxu) [inline edge/pkg/common/config/config.go]: remove defer func here

File: edge/pkg/devicetwin/config/config.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			if len(errs) != 0 {
+				for _, e := range errs {
+					klog.Errorf("%v", e)
+				}
+				klog.Error("init devicetwin config error")
+				os.Exit(1)
+			} else {
+				klog.Infof("init devicetwin config successfully，config info %++v", c)
+			}
+		}()
```
Comment (fisherxu) [inline edge/pkg/devicetwin/config/config.go]: ditto

File: edge/pkg/edged/config/config.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+			if len(errs) != 0 {
+				for _, e := range errs {
+					klog.Errorf("%v", e)
+				}
+				klog.Error("init edged config error")
+				os.Exit(1)
+			} else {
+				klog.Infof("init edged config successfully，config info %++v", c)
+			}
+		}()
```
Comment (fisherxu) [inline edge/pkg/edged/config/config.go]: ditto

File: edge/pkg/edged/config/config.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+			PodSandboxImage:           podSandboxImage,
+			ImagePullProgressDeadline: imagePullProgressDeadline,
+			RuntimeRequestTimeout: metav1.Duration{
+				Duration: time.Duration(runtimeRequestTimeout) * time.Minute,
+			},
+		}
+
+	})
+}
+func Get() Configure {
```
Comment (fisherxu) [inline edge/pkg/edged/config/config.go]: return pointer?

File: edge/pkg/edgehub/clients/factory.go [hunk 5]
```diff
@@ -20,7 +20,7 @@ const (
 var ErrorWrongClientType = errors.New("wrong Client Type")
 
 //GetClient returns an Adapter object with new web socket
-func GetClient(clientType string, config *config.EdgeHubConfig) (Adapter, error) {
+func GetClient(clientType string, config config.Configure) (Adapter, error) {
```
Comment (fisherxu) [inline edge/pkg/edgehub/clients/factory.go]: use pointer here

File: edge/pkg/edged/config/config.go (hunk 3 above)
Comment (kadisi) [reply edge/pkg/edged/config/config.go]: done

File: edge/pkg/devicetwin/config/config.go (hunk 2 above)
Comment (kadisi) [reply edge/pkg/devicetwin/config/config.go]: done

**Comment by kadisi** [comment]
@fisherxu @kevin-wangzefeng  PTAL

**Pull request merged**